hw/mcu/dialog/da1469x/src/hal_flash.c

```diff
@@ -382,5 +382,7 @@ da1469x_hff_sector_info(const struct hal_flash *dev, int idx,
 static int
 da1469x_hff_init(const struct hal_flash *dev)
 {
+    QSPIC->QSPIC_BURSTCMDA_REG = MYNEWT_VAL(QSPI_BURSTCMDA_INIT_VAL);
```

Comment:
there is potentially danger of changing those two values if code is already running from QSPI.
If for example for some reason it was running in single or dual mode, and function wants to change it to quad mode (or other way round) any interrupt between those two instructions could lead to crash. First register changes instruction that should be used while second set's up other vital parameters.
To be on the safe side code should be running from RAM with interrupts disabled.

Reply:
Ok, those are good points. It sounds like this really should be done only from flash_loader running in RAM, which is the only place right now where this is an issue. I'll close this and open a new PR.

Reply:
On second thought, I'll close this PR and open another.